web/modules/custom/form_api/src/Form/UserForm.php

```diff
@@ -0,0 +1,243 @@
+  }
+
+  /**
+   * Gets editable config file name.
+   */
+  public function getEditableConfigNames() {
+    return ['form_api.settings'];
+  }
+
+  /**
+   * Form constructor.
+   *
+   * Display a tree of all the form elements of user form.
+   *
+   * @param array $form
+   *   An associative array containing the structure of the form.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The current state of the form.
+   *
+   * @return array
+   *   The form structure.
+   */
+  public function buildForm(array $form, FormStateInterface $form_state = NULL) {
+    $config = $this->configFactory->get('form_api.settings');
+    $form = [];
+    $form['full_name'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Full Name'),
+      '#description' => $this->t('Enter your full name'),
+      '#required' => TRUE,
+      '#default_value' => $config->get('full_name') ?? '',
+    ];
+    $form['full-name-result'] = [
+      '#type' => 'markup',
+      '#markup' => "<div id='full-name-result' class='red'></div>",
+    ];
+    $form['phone_number'] = [
+      '#type' => 'tel',
+      '#title' => $this->t('Phone Number'),
+      '#description' => $this->t('Enter your phone number'),
+      '#required' => TRUE,
+      '#markup' => "<div id='phone-number-result'></div>",
+      '#default_value' => $config->get('phone_number') ?? '',
+    ];
+    $form['phone-number-result'] = [
+      '#type' => 'markup',
+      '#markup' => "<div id='phone-number-result' class='red'></div>",
+    ];
+    $form['email'] = [
+      '#type' => 'email',
+      '#title' => $this->t('Email Address'),
+      '#description' => $this->t('Enter your email address'),
+      '#markup' => "<div id='email-result'></div>",
+      '#default_value' => $config->get('email') ?? '',
+      '#required' => TRUE,
+    ];
+    $form['email-result'] = [
+      '#type' => 'markup',
+      '#markup' => "<div id='email-result' class='red'></div>",
+    ];
+    $form['gender'] = [
+      '#type' => 'radios',
+      '#title' => $this->t('Gender'),
+      '#description' => $this->t('Choose your gender'),
+      '#options' => [
+        'male' => $this->t('Male'),
+        'female' => $this->t('Female'),
+        'other' => $this->t('Other'),
+      ],
+      '#default_value' => $config->get('gender'),
+      '#required' => TRUE,
+    ];
+    $form['actions'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Submit'),
+      '#ajax' => [
+        'callback' => '::submitUsingAjax',
+        'progress' => [
+          'type' => 'throbber',
+          'message' => NULL,
+        ],
+      ],
+    ];
+
+    return $form;
+  }
+
+  /**
+   * Validates the form using AJAX and submits it.
+   *
+   * @param array $form
+   *   An associative array containing the structure of the form.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The current state of the form.
+   *
+   * @return \Drupal\Core\Ajax\AjaxResponse
+   *   The AJAX response.
+   */
```

Comment:
Same comment as above for this.

Reply:
The function submitUsingAjax is created by me and not an inherited function, it is called using ajax callback in $form['actions'].